app/jobs/scheduled/fix_user_usernames_and_groups_names_clash.rb

```diff
@@ -0,0 +1,29 @@
+module Jobs
+  class FixUserUsernamesAndGroupNamesClash < Jobs::Scheduled
```

Comment:
Why is this a recurring job and not a one-time (deferred) migration?

Reply:
We can't add any database index to ensure uniqueness at the database level. Therefore, there is still a chance a clash to happen.